docs/source/gateway/index.rst

```diff
@@ -1,6 +1,1049 @@
+
+Here are some examples for how you might use curl to interact with the Gateway:
+
+1. Getting information about a particular route: /routes/{name}
+This endpoint returns a serialized representation of the Route data structure.
+This provides information about the name and type, as well as the model details for the requested route endpoint.
+
+Sensitive configuration data from the server configuration file is not returned.
+
+.. code-block:: bash
+
+    curl -X GET http://my.gateway:8888/routes/embeddings
+
+2. Listing all configured routes on the Gateway: /routes
+
+This endpoint returns a list of all configured and initialized Route data for the MLflow AI Gateway server.
+
+.. code-block:: bash
+
+    curl -X GET http://my.gateway:8888/routes
+
+Sensitive configuration data from the server configuration file is not returned.
+
+3. Querying a particular route: /query/{route}
+This endpoint allows you to submit a query to a configured provider route. The data structure you send in the query depends on the route. Here are examples for the "completions", "chat", and "embeddings" routes:
+
+* ``Completions``
+
+.. code-block:: bash
+
+    curl -X POST http://my.gateway:8888/query/completions -H "Content-Type: application/json" -d '{"prompt": "Describe the probability distribution of first generation decay chain fission byproducts from a pressurized water nuclear reactor that uses 70% U-235 and 30% U-238"}'
+
+* ``Chat``
+
+.. code-block:: bash
+
+    curl -X POST http://my.gateway:8888/query/chat -H "Content-Type: application/json" -d '{"messages": [{"role": "user", "content": "Can you write a limerick about orange flavored popsicles?"}]}'
+
+* ``Embeddings``
+
+.. code-block:: bash
+
+    curl -X POST http://my.gateway:8888/query/embeddings -H "Content-Type: application/json" -d '{"texts": ["I'd like to return my shipment of beanie babies, please", "Can I please speak to a human now?"]}'
+
+These examples cover the primary ways you might interact with the MLflow AI Gateway via its REST API.
+
+**Note:** Please remember to replace "http://my.gateway:8888" with the URL of your actual MLflow AI Gateway Server.
+
+MLflow AI Gateway API Documentation
+===================================
 
 `API documentation <./api.html>`_
+
+
+.. _gateway-guide:
+
+Guide
+=====
+
+The following guide will assist you in getting up and running, using a 3-route configuration to
+OpenAI services for chat, completions, and embeddings.
+
+Step 1: Install the MLflow AI Gateway service
+---------------------------------------------
+First, you need to install the MLflow AI Gateway service on your machine. You can do this using pip from PyPI or from the MLflow repository.
+
+Installing from Pypi (recommended)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: sh
+
+    pip install 'mlflow[gateway]'
+
+
+Installing from a repository fork (for latest updates)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: sh
+
+    pip install -e '.[gateway]'
```

Comment:
We can remove this; power users will figure this out, and we don't want to encourage folks to install from master...stuff will probably break.